src/phoenix/server/api/routers/v1/datasets.py

```diff
@@ -25,19 +25,181 @@
+                )
+
+        query = query.limit(limit + 1)
+        result = await session.execute(query)
+        datasets = result.scalars().all()
+
+        if not datasets:
+            return JSONResponse(content={"next_cursor": None, "data": []}, status_code=200)
+
+        next_cursor = None
+        if len(datasets) == limit + 1:
+            next_cursor = str(GlobalID(NODE_NAME, str(datasets[-1].id)))
+            datasets = datasets[:-1]
+
+        data = []
+        for dataset in datasets:
+            data.append(
+                {
+                    "id": str(GlobalID(NODE_NAME, str(dataset.id))),
+                    "name": dataset.name,
+                    "description": dataset.description,
+                    "metadata": dataset.metadata_,
+                    "created_at": dataset.created_at.isoformat(),
+                    "updated_at": dataset.updated_at.isoformat(),
+                }
+            )
+
+        return JSONResponse(content={"next_cursor": next_cursor, "data": data})
+
+
+async def get_dataset_by_id(request: Request) -> Response:
+    """
+    summary: Get dataset by ID
+    operationId: getDatasetById
+    tags:
+      - datasets
+    parameters:
+      - in: path
+        name: id
+        required: true
+        schema:
+          type: string
+    responses:
+      200:
+        description: Success
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                id:
+                  type: string
+                name:
+                  type: string
+                description:
+                  type: string
+                metadata:
+                  type: object
+                created_at:
+                  type: string
+                  format: date-time
+                updated_at:
+                  type: string
+                  format: date-time
+                example_count:
+                  type: integer
+      404:
+        description: Dataset not found
+    """
+    dataset_id = GlobalID.from_id(request.path_params["id"])
+
+    if (type_name := dataset_id.type_name) != NODE_NAME:
+        return Response(
+            content=f"ID {dataset_id} refers to a f{type_name}", status_code=HTTP_404_NOT_FOUND
+        )
+    async with request.app.state.db() as session:
+        dataset = await session.get(models.Dataset, int(dataset_id.node_id))
+        if dataset is None:
+            return Response(
+                content=f"Dataset with ID {dataset_id} not found", status_code=HTTP_404_NOT_FOUND
+            )
+
+        output_dict = {
+            "id": str(dataset_id),
+            "name": dataset.name,
+            "description": dataset.description,
+            "metadata": dataset.metadata_,
+            "created_at": dataset.created_at.isoformat(),
+            "updated_at": dataset.updated_at.isoformat(),
+            "example_count": await dataset.load_example_count(session),
```

Comment:
How come we need to await a separate database/network call for the count? Wouldn't it be more efficient to combine it with the first one above by joining with a subquery?

Reply:
We don't need to do this, one query is more efficient but I also think the difference is very small once we have a connection

Reply:
If it's more efficient, then why not? Is the only reason that the difference is small?

Reply:
I feel this code is still liable to change as we work out where we want to push the complexity of querying for these objects (ORM Layer? A Pydantic model Layer? Entirely in the dataloaders?) Given that I'm writing the queries that are simple and make sense to me, and think we should make them most efficient once the structure and performance limitations become more clear

Reply:
In general I think we should aim to leverage SQL for what it's best suited for. I've never not combined queries and regretted it but I have not combined queries and inevitably had to revisit it a ton of times. In general let's combine queries whenever possible. It's the best way to hash out things early.